iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageManagerConsentFlagCheck.java

```diff
@@ -0,0 +1,105 @@
+import org.sonar.iac.docker.tree.api.Literal;
+import org.sonar.iac.docker.tree.api.RunInstruction;
+
+@Rule(key = "S6584")
+public class PackageManagerConsentFlagCheck implements IacCheck {
+  private static final String MESSAGE = "Add consent flag so that this command doesn't require user confirmation.";
+  private static final String ASSUME_YES_FLAG = "--assume-yes";
+  private static final Map<String, CommandOption> managersToConsentFlags = Map.of(
+    "apt-get", new CommandOption(List.of("-y"), List.of("--yes", ASSUME_YES_FLAG)),
+    "apt", new CommandOption(List.of("-y"), List.of("--yes", ASSUME_YES_FLAG)),
+    "aptitude", new CommandOption(List.of("-y"), List.of(ASSUME_YES_FLAG)));
+  private static final CommandDetector debianPackageManagerDetector = CommandDetector.builder()
+    .with(Set.of("apt", "apt-get", "aptitude"))
+    .withAnyFlag()
+    // these commands don't expect user confirmation even in interactive mode
+    .notWith(Set.of("update", "check", "changelog", "indextargets", "autoclean", "auto-clean")::contains)
+    .withOptionalRepeating(s -> true)
+    .build();
+
+  @Override
+  public void initialize(InitContext init) {
+    init.register(RunInstruction.class, PackageManagerConsentFlagCheck::checkRunInstruction);
+  }
+
+  private static void checkRunInstruction(CheckContext ctx, RunInstruction runInstruction) {
+    List<ArgumentResolution> argumentResolutions = CheckUtils.resolveInstructionArguments(runInstruction);
+    debianPackageManagerDetector.search(argumentResolutions).forEach(c -> {
+      if (isPackageManagerInvocationWithoutConsent(c)) {
+        ctx.reportIssue(c.textRange(), MESSAGE);
+      }
+    });
+
+  }
+
+  private static boolean isPackageManagerInvocationWithoutConsent(CommandDetector.Command command) {
+    String commandName = command.getResolvedArguments().get(0).value();
+
+    return command.getResolvedArguments().stream()
+      .filter(resolution -> resolution.argument().expressions().get(0) instanceof Literal)
+      .map(resolution -> (Literal) resolution.argument().expressions().get(0))
+      .noneMatch(literal -> managersToConsentFlags.get(commandName).matches(literal.value()));
+  }
+
+  static class CommandOption {
+    final List<String> shortVariants;
+    final List<String> longVariants;
+
+    CommandOption(List<String> shortVariants, List<String> longVariants) {
+      this.shortVariants = shortVariants;
+      this.longVariants = longVariants;
+    }
+
+    boolean matches(String flag) {
+      if (flag.startsWith("--")) {
+        for (String longVariant : longVariants) {
+          if (longVariant.equals(flag)) {
+            return true;
+          }
+        }
+        return false;
+      }
+
+      for (String shortVariant : shortVariants) {
+        // this takes into account that short options can be glued together
+        // this way we will e.g. detect `y` in `-tmy`
+        if (flag.indexOf(shortVariant.charAt(1)) != -1) {
```

Comment:
maybe this will be better: `flag.startsWith("-"") && flag.contains(shortVariant.charAt(1))`?

Reply:
Thanks. I'll add the first part. But `contains` doesn't accept single characters, unfortunately..